airbyte-webapp/src/views/Connector/ConnectorCard/ConnectorCard.tsx

```diff
@@ -1,31 +1,51 @@
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 import { FormattedMessage } from "react-intl";
 
 import { JobItem } from "components/JobItem/JobItem";
 import { Card } from "components/ui/Card";
 
-import { Connector, ConnectorSpecification, ConnectorT } from "core/domain/connector";
+import {
+  Connector,
+  ConnectorDefinition,
+  ConnectorDefinitionSpecification,
+  ConnectorSpecification,
+  ConnectorT,
+} from "core/domain/connector";
 import { SynchronousJobRead } from "core/request/AirbyteClient";
 import { LogsRequestError } from "core/request/LogsRequestError";
 import { useAdvancedModeSetting } from "hooks/services/useAdvancedModeSetting";
 import { generateMessageFromError } from "utils/errorStatusMessage";
-import { ServiceForm, ServiceFormProps, ServiceFormValues } from "views/Connector/ServiceForm";
+import { ConnectorFormValues, ServiceForm, ServiceFormValues } from "views/Connector/ServiceForm";
 
+import { useDocumentationPanelContext } from "../ConnectorDocumentationLayout/DocumentationPanelContext";
 import { ConnectorServiceTypeControl } from "../ServiceForm/components/Controls/ConnectorServiceTypeControl";
 import styles from "./ConnectorCard.module.scss";
 import { useAnalyticsTrackFunctions } from "./useAnalyticsTrackFunctions";
 import { useTestConnector } from "./useTestConnector";
 
-type ConnectorCardProvidedProps = Omit<
-  ServiceFormProps,
-  "isKeyConnectionInProgress" | "isSuccess" | "onStopTesting" | "testConnector"
->;
-
-interface ConnectorCardBaseProps extends ConnectorCardProvidedProps {
+// TODO: need to clean up the ConnectorCard and ServiceForm props,
+// since some of props are used in both components, and some of them used just as a prop-drill
+interface ConnectorCardBaseProps {
   title?: React.ReactNode;
   full?: boolean;
   jobInfo?: SynchronousJobRead | null;
   additionalSelectorComponent?: React.ReactNode;
+  onSubmit: (values: ConnectorFormValues) => Promise<void> | void;
+  onServiceSelect?: (id: string) => void;
+  availableServices: ConnectorDefinition[];
```

Comment:
While we're here changing things anyway, we should go ahead and try to get rid of the term "service" from all of these places. 

It is confusing that this component and the ServiceForm component are the only places that refer to connectors and connector definitions as "services", so we should fix that. In this case, better names for these props would be `onConnectorDefinitionSelect` and `availableConnectorDefinitions`, to better match the terminology that we use for these objects throughout the rest of the Airbyte codebase.

Ideally we would also rename `ServiceForm` to something like `ConnectorForm`, and rename all related methods/variables/components as such.

Reply:
Also was thinking about that, but wasn't sure 😊
Ok, will fix that!

Reply:
Decided to try to rename all "service" related usages - got more modified files than I expected 😕 
Good news - no "service" term mentions anymore 😊